aiida_testing/mock_code/_fixtures.py

```diff
@@ -71,6 +79,44 @@ def mock_fail_on_missing(request):
     return request.config.getoption("--mock-fail-on-missing")
 
 
+def _mock_get_option(self, name: str):
+    """
+    Return the value of an option that was set for this CalcJobNode
+
+    Always return `False` if option is `withmpi`.
+
+    :param name: the option name
+    :return: the option value or None
+    :raises: ValueError for unknown option
+    """
+    if name == 'withmpi':
+        return False
+    return self.base.attributes.get(name, None)
+
+
+@pytest.fixture(scope='function')
+def mock_disable_mpi(request, monkeypatch):
+    """Enforce `withmpi=False` based on `--mock-disable-mpi` cli option.
+
+    This is achieved by monkey-patching the `CalcJob.get_option()` method.
+    """
+    if not request.config.getoption("--mock-disable-mpi"):
+        return
+
+    from aiida.orm import CalcJobNode  # pylint: disable=import-outside-toplevel
+    monkeypatch.setattr(CalcJobNode, 'get_option', _mock_get_option)
+
+
+@pytest.fixture(scope='function')
+def mock_disable_mpi_force(monkeypatch):
+    """Enforce `withmpi=False` independent of `--mock-disable-mpi` cli option.
+
+    Mostly required to test the functionality.
```

Comment:
If this is only for internal use, maybe we should rename it to `_mock_disable_mpi_force`?